### PR TITLE
Do not fail for w<=0 in Complex TransferFunction

### PR DESCRIPTION
**Pull request by HansOlsson**
Trying to simulate TransferFunction for w=0 fails because (j*w)^(i-1) tries to compute a complex power which fails for zero exponent.
This change corrects that in two different ways - we could decide to do just one of them.
1. By handling it in TransferFunction - There is an extra protected function, but it could be removed by some loss of readability.
2. By introducing Complex ^Integer in the complex class; note that by using the type of the exponent we avoid the odd cases in a safe way.

### CODE REVIEW

File: Complex.mo [hunk 1]
```diff
@@ -166,21 +166,35 @@ operator record Complex "Complex number with overloaded operators"
...
+    end complexPower;
+    function integerPower "Integer power of complex number"
+      import Complex;
+      input Complex c1 "Complex number";
+      input Integer c2 "Integer exponent";
+      output Complex c3 "= c1^c2";
+    algorithm
+      c3 := if c2==0 then Complex(1,0) else Complex.'^'.complexPower(c1,Complex(c2,0));
+      annotation(Inline=true, smoothOrder=100, Documentation(info="<html>
+<p>This function returns the given Complex numbers c1 to the power of the Integer number c2.</p>
```
Comment (beutlich) [inline Complex.mo:194]:
```suggestion
<p>This function returns the given Complex number c1 to the power of the Integer number c2.</p>
```

Probably also for complexPower.

File: Complex.mo [hunk 2]
```diff
@@ -166,21 +166,35 @@ operator record Complex "Complex number with overloaded operators"
...
 <p>This function returns the given Complex numbers c1 to the power of the Complex number c2.</p>
 </html>"));
+    end complexPower;
+    function integerPower "Integer power of complex number"
+      import Complex;
+      input Complex c1 "Complex number";
+      input Integer c2 "Integer exponent";
+      output Complex c3 "= c1^c2";
+    algorithm
+      c3 := if c2==0 then Complex(1,0) else Complex.'^'.complexPower(c1,Complex(c2,0));
```
Comment (beutlich) [inline Complex.mo:192]: Two comments on `Complex(c2,0)`
* It uses the fromReal ctor which is an implicit cast from Integer to Real, which is OK is guess.
* The second argument can be skipped.

File: Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo [hunk 3]
```diff
@@ -15,9 +15,16 @@ protected
   Complex aw[size(a,1)];
   Complex bSum;
   Complex aSum;
+  function PowerOfI "Internal function to compute j^k for any integer"
```
Comment (beutlich) [inline Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo:18]: Function names should start with a lower-case letter.

Should it be an encapsulated function?

File: Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo [hunk 4]
```diff
@@ -15,9 +15,16 @@ protected
...
+    input Integer k;
+    output Complex x;
+  algorithm
+   x:=if mod(k,4)==1 then Complex(1,0) elseif mod(k,4)==2 then j elseif mod(k,4)==3 then Complex(-1,0) else -j;
```
Comment (beutlich) [inline Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo:22]: Maybe use an internal variable to store mod(k,4) and explicitly set the Inline=true annotation.

The second argument of Complex ctor can be skipped.

File: Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo (hunk 3 above)
Comment (HansOlsson) [reply Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo:18]: I will change first letter. I don't think it should be encapsulated, but perhaps moved to another place in the future.

File: Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo (hunk 4 above)
Comment (HansOlsson) [reply Modelica/ComplexBlocks/ComplexMath/TransferFunction.mo:22]: Ok, did that - and corrected an error that didn't influence the end-results as well.

**Review by christiankral — COMMENTED** [review]
As we use `j` not `i` of the complex notation in Modelica, I wonder if shall call the function `PowerOfJ` instead of `PowerOfI`

**Review by christiankral — APPROVED** [review]
Looks very good

**Comment by christiankral** [comment]
We have keep in mind to later consider #3651 and modify the PR #3733 accordingly

**Review by MartinOtter — APPROVED** [review]

**Pull request merged**